saleor/graphql/account/mutations.py

```diff
@@ -453,10 +455,38 @@ def user_is_allowed(cls, user, input):
         return user.has_perm('account.manage_users')
 
 
+class CustomerAddressCreate(ModelMutation):
+    class Arguments:
+        input = AddressInput(
+            description='Fields required to create address', required=True)
+
+    class Meta:
+        description = 'Creates address for user'
+        model = models.Address
+        exclude = ['user_addresses']
+
+    @classmethod
+    def user_is_allowed(cls, user, input):
+        return user.is_authenticated
+
+    @classmethod
+    def save(cls, info, instance, cleaned_input):
+        user = info.context.user
+        super().save(info, instance, cleaned_input)
+        instance.user_addresses.add(user)
+        instance.save()
+
+
+def check_is_stuff_or_user_edit_own_address(info, instance):
+    user = info.context.user
+    if not (user.has_perm('account.manage_users')
+            or user in instance.user_addresses.all()):
```

Comment:
I'm not sure getting all the user addresses is a good solution. It's pretty heavy to do that check such way.

Reply:
I've refactored it a little bit but I'd say that this is acceptable.